app/src/main/java/com/framgia/bitcoinwallet/ui/screen/main/sendcointab/SendCoinFragment.kt

```diff
@@ -1,26 +1,86 @@
+import android.app.AlertDialog
+import android.arch.lifecycle.Observer
 import android.view.View
+import com.framgia.bitcoinwallet.databinding.FragmentSendCoinBinding
+import kotlinx.android.synthetic.main.fragment_send_coin.view.*
+import android.os.Build
+import android.widget.Toast
 import com.framgia.bitcoinwallet.R
 import com.framgia.bitcoinwallet.ui.BaseFragment
+import com.framgia.bitcoinwallet.ui.screen.main.MainActivity
+import com.framgia.bitcoinwallet.util.obtainViewModel
 
-class SendCoinFragment: BaseFragment() {
+class SendCoinFragment : BaseFragment<FragmentSendCoinBinding>(), SendCoinNavigator {
 
-    override fun initComponentsOnCreate(savedInstanceState: Bundle?) {
+    companion object {
+        fun newInstance() = SendCoinFragment()
+        private const val TAG = "SendCoinFragment"
+    }
 
+    override fun getLayoutRes() = R.layout.fragment_send_coin
+
+    override fun initData() {
+        viewDataBinding.apply {
+            viewModel = (activity as MainActivity).obtainViewModel(SendCoinViewModel::class.java)
+            viewModel?.let { lifecycle.addObserver(it) }
+        }
     }
 
-    override fun getLayoutRes(): Int {
-        return R.layout.fragment_send_coin
+    override fun startScanScreen() {
+
     }
 
-    override fun initComponentsOnCreateView(rootView: View, savedInstanceState: Bundle?) {
+    override fun observeModelData(view: View) {
+        viewDataBinding?.viewModel?.showAlertDialog?.observe(this, Observer {
+            if (it != null && it) {
+                showVerifySendDialog(view)
+            }
+        })
 
+        viewDataBinding?.viewModel?.sendCoinState?.observe(this, Observer {
+            if (it != null && it) {
+                notifyMessage(getString(R.string.send_coin_successed))
+            } else {
+                notifyMessage(getString(R.string.send_coin_failed))
+            }
+        })
     }
 
-    companion object {
-        fun newInstance() = SendCoinFragment()
-        private const val TAG = "SendCoinFragment"
+    override fun setEvents(view: View) {
+        view.button_send_coin.setOnClickListener {
+            viewDataBinding?.viewModel?.verifySendCoin(view.edit_bitcoin_add.text.toString(),
+                    view.edit_amount.text.toString())
+        }
+
+        view.image_scan.setOnClickListener { startScanScreen() }
+    }
+
+    private fun showVerifySendDialog(view: View) {
+        val builder: AlertDialog.Builder
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            builder = AlertDialog.Builder(context, android.R.style.Theme_Material_Dialog_Alert)
+        } else {
+            builder = AlertDialog.Builder(context)
+        }
+        builder.setTitle(getString(R.string.dialog_send_title))
+                .setMessage(getString(R.string.dialog_send_message))
+                .setPositiveButton(android.R.string.yes) { dialog, which ->
+                    viewDataBinding?.viewModel?.showAlertDialog?.value = false
```

Comment:
viewBinbing của em not null rồi mà, ko cần ? nữa
hình như android stuiod có báo chứ nhỉ

Reply:
à vâng, e quen tay cứ check, để em sửa ạ